gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/tasks/CheckPegasusSnapshotTask.java

```diff
@@ -173,4 +190,53 @@ public void setHandlerJarPath(FileCollection handlerJarPath)
   {
     _handlerJarPath = handlerJarPath;
   }
+
+  private boolean hasSchemaAnnotationHandler()
+  {
+    int expectedHandlersNumber = ((DefaultConfiguration) _handlerJarPath).getAllDependencies().size();
+    // skip if no handlers configured
+    if (expectedHandlersNumber == 0)
+    {
+      getLogger().info("no schema annotation handlers configured for schema annotation compatibility check");
+      return false;
+    }
+
+    List<URL> handlerJarPathUrls = new ArrayList<>();
+
+    for (File f : _handlerJarPath)
+    {
+      try
+      {
+        handlerJarPathUrls.add(f.toURI().toURL());
+      }
+      catch (MalformedURLException e)
+      {
+        new GradleException("Annotation compatibility check: could not get schema annotation handler jar " + f.getName() + " to url. " + e.getMessage());
+      }
+    }
+
+    ClassLoader classLoader = new URLClassLoader(handlerJarPathUrls.toArray(new URL[handlerJarPathUrls.size()]),
+        getClass().getClassLoader());
+
+    try
+    {
+      _handlerClassNames = SchemaAnnotationHandlerClassUtil.getAnnotationHandlerClassNames(_handlerJarPath, classLoader, getProject());
+    }
+    catch (IOException e)
+    {
+      new GradleException("Annotation compatibility check: could not get annotation handler class name. " + e.getMessage());
+    }
+
+    // For now, every schema annotation handler should be in its own module
```

Comment:
The logic is still similar to ValidateSchemaAnnotationTask, do you think we can extract the common part to not repeat yourself? Later if we change the enforcement of expected number of hanlders per module, we don't need to change on two places

Reply:
You are right, I will move this part to the SchemaAnnotationHandlerUtil.